src/pool.rs

```diff
@@ -149,11 +230,37 @@ impl ConnectionPool {
         }
     }
 
-    // Connects to a MongoDB server as defined by the initial configuration.
-    fn connect(&self) -> Result<BufStream<TcpStream>> {
+
+    // Connects to a MongoDB server as defined by the initial
+    #[allow(unreachable_code)] // Suppresses warning for `panic` when ssl is enabled.
+    fn connect(&self) -> Result<BufStream<Stream>> {
         let host_name = &self.host.host_name;
         let port = self.host.port;
-        let stream = BufStream::new(try!(TcpStream::connect((&host_name[..], port))));
-        Ok(stream)
+        let inner_stream = try!(TcpStream::connect((&host_name[..], port)));
+
+        if let Some(SslConfig { ca_file: ref _ca_file,
+                                certificate_file: ref _certificate_file,
+                                key_file: ref _key_file }) = self.ssl {
+            #[cfg(feature = "ssl")]
```

Comment:
heyo, re: this, I'm imagining moving this into its own module and having `Stream` be a Trait instead of an enum that's matched on. Something similar to what's being down in [rust-postgres](https://github.com/sfackler/rust-postgres/tree/master/postgres/src/tls). This would move most of the ssl-specific code into a single module.